bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -4885,20 +4885,18 @@ LRESULT WM_DESTROY (long wParam, long lParam) {
 LRESULT WM_DPICHANGED (long wParam, long lParam) {
 	// Map DPI to Zoom and compare
 	int nativeZoom = DPIUtil.mapDPIToZoom (OS.HIWORD (wParam));
-	int newSWTZoom = DPIUtil.getZoomForAutoscaleProperty (nativeZoom);
-	int oldSWTZoom = getShell().getZoom();
+	int oldNativeZoom = getShell().getNativeZoom();
 
 	// Throw the DPI change event if zoom value changes
-	if (newSWTZoom != oldSWTZoom) {
+	if (nativeZoom != oldNativeZoom) {
```

Comment:
Just a remark about this line, so everyone is aware of it. Before the deviceZoom was checked, so for the default value for swt.autoScale (integer200) a zoom change from 125 to 150 was converted to newSWTZoom and oldSWTZoom  both being 100, so no zoom event was propagated. Now, as the native zoom is used, this condition will be true and the event will be propagated. As fonts are based on the native zoom, they will be adapted. I this this behaviour is more consistent, but wanted to point it out.

Reply:
Thanks for pointing that out! Interesting observation and probably something that would have produced bugs later on.